addons/token-exchange/token_exchanger_agent.go

```diff
@@ -153,22 +154,53 @@ func (c *tokenExchangeAgentController) sync(ctx context.Context, syncCtx factory
 		return nil
 	}
 
-	se, err := c.spokeSecretLister.Secrets(namespace).Get(name)
+	c.synHubSecrets(name, namespace)
+
+	// TODO: Read managed cluster secrets
+	return nil
+}
+
+// synHubSecrets syncs secrets from hub to spoke
+func (c *tokenExchangeAgentController) synHubSecrets(selector, namespace string) error {
+	return c.synSecrets(c.hubSecretLister, selector, namespace)
+}
+
+func (c *tokenExchangeAgentController) synSecrets(lister corev1lister.SecretLister, selector, namespace string) error {
```

Comment:
This can be reused if it accepts source and destination information.

Reply:
done

Reply:
done